src/PaymentMethods/Billie.php

```diff
@@ -35,11 +35,77 @@ protected function getConfig(): array
         ];
     }
 
+    public function filtersOnBuild()
+    {
+        add_filter(
+            'woocommerce_after_checkout_validation',
+            [$this, 'BillieFieldsMandatory'],
+            11,
+            2
+        );
+        add_action(
+            'woocommerce_checkout_posted_data',
+            [$this, 'switchFields'],
+            11
+        );
+    }
+
     public function getFormFields($generalFormFields): array
     {
         unset($generalFormFields[1]);
         unset($generalFormFields['allowed_countries']);
 
         return $generalFormFields;
     }
+
+    public function BillieFieldsMandatory($fields, $errors)
+    {
+        $gatewayName = "mollie_wc_gateway_billie";
+        $field = 'billing_company_billie';
+        $companyLabel = __('Company', 'mollie-payments-for-woocommerce');
+        return $this->addPaymentMethodMandatoryFields($fields, $gatewayName, $field, $companyLabel, $errors);
+    }
+
+    public function switchFields($data)
```

Comment:
is it because of the interface that most of the methods here don't use type hints or return types?